### PR TITLE
datalad: 1.0.2 -> 1.1.1, add tests

### PR DESCRIPTION
**Pull request by gador**
## Description of changes

- Update to most recent version [1.1.1](https://github.com/datalad/datalad/blob/maint/CHANGELOG.md)
- Add tests

ping @renesat

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by gador** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).

Result of `nixpkgs-review pr 329398` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
6 packages built:
</summary>
<ul>
<li>datalad</li>
<li>datalad.dist</li>
<li>python311Packages.heudiconv</li>
<li>python311Packages.heudiconv.dist</li>
<li>python312Packages.heudiconv</li>
<li>python312Packages.heudiconv.dist</li>
</ul>
</details>

**Comment by gador** [comment]
Result of `nixpkgs-review pr 329398` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>6 packages built:</summary>
  <ul>
    <li>datalad</li>
    <li>datalad.dist</li>
    <li>python311Packages.heudiconv</li>
    <li>python311Packages.heudiconv.dist</li>
    <li>python312Packages.heudiconv</li>
    <li>python312Packages.heudiconv.dist</li>
  </ul>
</details>

File: pkgs/applications/version-management/datalad/default.nix [hunk 1]
```diff
@@ -72,13 +90,120 @@ python3.pkgs.buildPythonApplication rec {
...
+    "test_get_cached_dataset"
+    "test_cached_dataset"
+    "test_cached_url"
+    "test_anonymous_s3"
+    "test_protocols"
+    "test_get_versioned_url_anon"
+    "test_install_recursive_github"
+    "test_failed_install_multiple"
+
+    # hardcoded /bin/ls path in "test"
```
Comment (renesat) [inline pkgs/applications/version-management/datalad/default.nix:182]: You can easily patch it as I understand. Maybe use `substituteInPlace`

File: pkgs/applications/version-management/datalad/default.nix [hunk 2]
```diff
@@ -1,68 +1,86 @@
...
 
 python3.pkgs.buildPythonApplication rec {
   pname = "datalad";
-  version = "1.0.2";
+  version = "1.1.1";
 
   src = fetchFromGitHub {
     owner = "datalad";
     repo = pname;
     rev = "refs/tags/${version}";
```
Comment (renesat) [inline pkgs/applications/version-management/datalad/default.nix:20]:
```suggestion
    rev = version;
```

File: pkgs/applications/version-management/datalad/default.nix [hunk 3]
```diff
@@ -1,68 +1,86 @@
...
-    distro
-  ] ++ lib.optionals stdenv.hostPlatform.isWindows [ colorama ]
+  nativeBuildInputs = [
+    installShellFiles
+    git
+  ];
+
+  build-system = [ python3.pkgs.setuptools ];
+
+  dependencies =
```
Comment (renesat) [inline pkgs/applications/version-management/datalad/default.nix:31]: I think we need to switch dependencies. They've changed them a lot: https://github.com/datalad/datalad/blob/0b18b32ee05cf6d7d12772719458ae5526d52553/setup.py#L53

Comment (gador) [reply pkgs/applications/version-management/datalad/default.nix:31]: They did change a bit. I adjusted it (removed  `simplejson` and `whoosh`).

File: pkgs/applications/version-management/datalad/default.nix (hunk 1 above)
Comment (gador) [reply pkgs/applications/version-management/datalad/default.nix]: good point, done

File: pkgs/applications/version-management/datalad/default.nix (hunk 2 above)
Comment (gador) [reply pkgs/applications/version-management/datalad/default.nix:21]: no, see https://discourse.nixos.org/t/why-is-the-r-ryantm-bot-using-refs-tags-in-the-version/20519

File: pkgs/applications/version-management/datalad/default.nix (hunk 3 above)
Comment (renesat) [reply pkgs/applications/version-management/datalad/default.nix:37]: Yes, but for example pyyaml, mutagen and others do not exist in `setup.py`. You need to remove them.

File: pkgs/applications/version-management/datalad/default.nix [hunk 4]
```diff
@@ -1,69 +1,91 @@
...
+      argcomplete
+      pyperclip
+      python-dateutil
+
+      # metadata-extra
+      pyyaml
+      mutagen
+      exifread
+      python-xmp-toolkit
+      pillow
```
Comment (renesat) [inline pkgs/applications/version-management/datalad/default.nix:77]: You add some packages, but for example pyyaml, mutagen and others do not exist in `setup.py`. You need to remove them.

**Review by renesat — COMMENTED** [review]
Please don't mark comments as resolved. It's so hard for me to respond to them.

File: pkgs/applications/version-management/datalad/default.nix (hunk 4 above)
Comment (gador) [reply pkgs/applications/version-management/datalad/default.nix:77]: Oh yes, I must've overlooked them. Resolved.

**Review by renesat — APPROVED** [review]
Sorry for the delay. I think all is well now and we can merge.

**Comment by gador** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).

Result of `nixpkgs-review pr 329398` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
6 packages built:
</summary>
<ul>
<li>datalad</li>
<li>datalad.dist</li>
<li>python311Packages.heudiconv</li>
<li>python311Packages.heudiconv.dist</li>
<li>python312Packages.heudiconv</li>
<li>python312Packages.heudiconv.dist</li>
</ul>
</details>

**Comment by gador** [comment]
Result of `nixpkgs-review pr 329398` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>6 packages built:</summary>
  <ul>
    <li>datalad</li>
    <li>datalad.dist</li>
    <li>python311Packages.heudiconv</li>
    <li>python311Packages.heudiconv.dist</li>
    <li>python312Packages.heudiconv</li>
    <li>python312Packages.heudiconv.dist</li>
  </ul>
</details>

**Pull request merged**